client/src/components/ViewPostDialog.tsx

```diff
@@ -123,7 +126,9 @@ function MoreOptions(props: {
         ) : (
           <></>
         )}
-        <MenuItem onClick={reportPost}>Report</MenuItem>
+        {!props.didUserReport ? (
```

Comment:
Just a nit, I don't think we should hide it if the post is already reported -- maybe just say "Post has already been reported". On other sites, if you report a post, the button is not hidden.